pkg/resources/view.go

```diff
@@ -52,172 +59,550 @@ var viewSchema = map[string]*schema.Schema{
+			statement, err := extractor.Extract()
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			if err = d.Set("statement", statement); err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			return diag.FromErr(fmt.Errorf("error reading view %v, err = %w, `text` is missing; if the view is secure then the role used by the provider must own the view (consult https://docs.snowflake.com/en/sql-reference/sql/create-view#usage-notes)", d.Id(), err))
 		}
-	} else {
-		return fmt.Errorf("error reading view %v, err = %w, `text` is missing; if the view is secure then the role used by the provider must own the view (consult https://docs.snowflake.com/en/sql-reference/sql/create-view#usage-notes)", d.Id(), err)
+
+		describeResult, err := client.Views.Describe(ctx, view.ID())
+		if err != nil {
+			log.Printf("[DEBUG] describing view: %s, err: %s", id.FullyQualifiedName(), err)
+		} else {
+			if err = d.Set(DescribeOutputAttributeName, schemas.ViewDescriptionToSchema(describeResult)); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		if err = d.Set(ShowOutputAttributeName, []map[string]any{schemas.ViewToSchema(view)}); err != nil {
+			return diag.FromErr(err)
+		}
+		return nil
 	}
+}
 
-	return nil
+func handlePolicyReferences(ctx context.Context, client *sdk.Client, id sdk.SchemaObjectIdentifier, d *schema.ResourceData) error {
+	// ensure a warehouse is selected for the session to get policy references
+	cleanupWarehouse, err := ensureWarehouse(ctx, client)
+	if err != nil {
+		return err
+	}
+	defer cleanupWarehouse()
+
+	policyRefs, err := client.PolicyReferences.GetForEntity(ctx, sdk.NewGetForEntityPolicyReferenceRequest(id, sdk.PolicyEntityDomainView))
+	if err != nil {
+		return err
+	}
+	for _, p := range policyRefs {
+		policyName := sdk.NewSchemaObjectIdentifier(*p.PolicyDb, *p.PolicySchema, p.PolicyName)
+		switch p.PolicyKind {
+		case string(sdk.PolicyKindAggregationPolicy):
+			var entityKey []string
+			if p.RefArgColumnNames != nil {
+				entityKey = sdk.ParseCommaSeparatedStringArray(*p.RefArgColumnNames, true)
+			}
+			if err = d.Set("aggregation_policy", []map[string]any{{
+				"policy_name": policyName.FullyQualifiedName(),
+				"entity_key":  entityKey,
+			}}); err != nil {
+				return err
+			}
+		case string(sdk.PolicyKindRowAccessPolicy):
+			var on []string
+			if p.RefArgColumnNames != nil {
+				on = sdk.ParseCommaSeparatedStringArray(*p.RefArgColumnNames, true)
+			}
+			if err = d.Set("row_access_policy", []map[string]any{{
+				"policy_name": policyName.FullyQualifiedName(),
+				"on":          on,
+			}}); err != nil {
+				return err
+			}
+		default:
+			log.Printf("[WARN] unexpected policy kind %v in policy references returned from Snowflake", p.PolicyKind)
+		}
+	}
+	return err
+}
+
+type viewWithReplaceDataResourceGetter struct {
+	d *schema.ResourceData
+}
+
+func (g *viewWithReplaceDataResourceGetter) Get(name string) any {
```

Comment:
I thought we were going to handle only one field change per `HasChange`, so I decided to use old values for handling the changed statement. Now, I altered it so that we recreate the resource with new values.